packages/medusa/src/services/order-edit.ts

```diff
@@ -0,0 +1,50 @@
+import { EntityManager } from "typeorm"
+import { MedusaError } from "medusa-core-utils"
+import { OrderEditRepository } from "../repositories/order-edit"
+import { OrderEditStatus } from "../models/order-edit"
+import { TransactionBaseService } from "../interfaces"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  orderEditRepository: typeof OrderEditRepository
+}
+
+export class OrderEditService extends TransactionBaseService {
+  protected transactionManager_: EntityManager | undefined
+  protected readonly manager_: EntityManager
+
+  protected readonly orderEditRepository_: typeof OrderEditRepository
+
+  constructor({ manager, orderEditRepository }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.orderEditRepository_ = orderEditRepository
+  }
+
+  async delete(orderEditId: string): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEditRepo = manager.getCustomRepository(
+        this.orderEditRepository_
+      )
+
+      const edit = await orderEditRepo.findOne({ where: { id: orderEditId } })
+
+      if (!edit) {
+        return
+      }
+
+      if (edit.status !== OrderEditStatus.CREATED) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Cannot delete order edit with status ${edit.status}`
+        )
+      }
+
+      await orderEditRepo.softRemove(edit)
```

Comment:
thought(non-blocking): Given that OrderEdits in their created state are without effect, one could argue that performing a `remove` would be reasonable. Obviously, will prevent us from showing it as an event in the change history (the event is not super informing in this state, so should be fine), but on the other hand, would free up DB storage - even though it might not be a significant amount.

Reply:
@olivermrbl I am pretty agree, I would add that since an order edit can only be removed at that state and if we go for a complete remove, would it make sense to make the entity not soft deletable as there is no other state where we want to really remove it?

Reply:
Yeah, if we choose to use remove, we should use a soft deletable entity calss.

Reply:
I think that if we hard remove then it should not extend the soft deletable class no? 😅

Reply:
Sorry, that's what I meant 😆

Reply:
Ahah alright, i thought i missed something for a minute 😄

Reply:
I will open a new pr for this 😄